addons/account/static/src/js/reconciliation/reconciliation_model.js

```diff
@@ -230,10 +230,10 @@ var StatementModel = BasicModel.extend({
         var line = this.getLine(handle);
         line.st_line.partner_id = partner && partner.id;
         line.st_line.partner_name = partner && partner.display_name || '';
-        return this._changePartner(handle, partner.id)
+        return $.when(partner && this._changePartner(handle, partner.id))
                 .then(function() {
                     line.reconciliation_proposition = [];
-                    return self._performMoveLine(handle)
+                    return self.changeMode(handle, 'match');
```

Comment:
good to me, related to a task that asked to set mode in match when changing partner